app/src/main/res/layout/adapter_kind_items.xml

```diff
@@ -9,61 +9,81 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toTopOf="parent" />
 
     <TextView
             android:id="@+id/ItemName"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
+            android:layout_width="225dp"
+            android:layout_height="36dp"
+            android:layout_marginStart="51dp"
+            android:layout_marginTop="16dp"
+            android:layout_marginEnd="54dp"
+            android:layout_marginBottom="56dp"
+            android:ellipsize="end"
             android:fontFamily="@font/shellypalmer_circular_std_book"
+            android:maxLines="1"
+            android:scrollHorizontally="true"
             android:text="Stall Name"
             android:textAlignment="center"
             android:textColor="@color/colorWhite"
-            android:textSize="24sp"
+            android:textSize="22sp"
             android:textStyle="bold"
-            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintBottom_toBottomOf="@+id/stallImage"
             app:layout_constraintEnd_toEndOf="@+id/stallImage"
-            app:layout_constraintStart_toStartOf="@+id/stallImage"
+            app:layout_constraintStart_toEndOf="@+id/Stall_photo"
             app:layout_constraintTop_toTopOf="@+id/stallImage" />
 
     <TextView
             android:id="@+id/price"
-            android:layout_width="30dp"
-            android:layout_height="0dp"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="169dp"
+            android:layout_marginTop="56dp"
+            android:layout_marginBottom="16dp"
             android:fontFamily="@font/shellypalmer_circular_std_book"
             android:gravity="center"
             android:text="0"
             android:textAlignment="center"
             android:textColor="@color/colorWhite"
-            android:textSize="24sp"
-            android:textStyle="bold"
-            app:layout_constraintBottom_toBottomOf="@+id/ItemName"
-            app:layout_constraintEnd_toEndOf="@+id/stallImage"
-            app:layout_constraintHorizontal_bias="0.6"
-            app:layout_constraintStart_toEndOf="@+id/ItemName"
-            app:layout_constraintTop_toTopOf="@+id/ItemName" />
+            android:textSize="22sp"
+            app:layout_constraintBottom_toBottomOf="@+id/stallImage"
+            app:layout_constraintStart_toStartOf="@+id/stallImage"
+            app:layout_constraintTop_toTopOf="@+id/stallImage" />
 
     <ImageView
-            android:id="@+id/kImg"
-            android:layout_width="56dp"
-            android:layout_height="50dp"
-            app:layout_constraintBottom_toBottomOf="@+id/ItemName"
-            app:layout_constraintEnd_toStartOf="@+id/ItemName"
-            app:layout_constraintHorizontal_bias="0.28"
+            android:id="@+id/Stall_photo"
```

Comment:
According to the convention followed throughout the codebase, widget ids should start with lower case